src/function/list/list_prepend_function.cpp

```diff
@@ -28,21 +30,41 @@ struct ListPrepend {
     }
 };
 
+static void validateArgumentType(const binder::expression_vector& arguments) {
+
+    if (binder::ExpressionUtil::isEmptyList(*arguments[0]))
+        return;
+
+    if (arguments[0]->getDataType().getLogicalTypeID() != LogicalTypeID::ANY &&
+        arguments[1]->dataType != ListType::getChildType(arguments[0]->dataType)) {
+        throw BinderException(
+            ExceptionMessage::listFunctionIncompatibleChildrenType(ListPrependFunction::name,
+                arguments[0]->getDataType().toString(), arguments[1]->getDataType().toString()));
+    }
+}
+
 static std::unique_ptr<FunctionBindData> bindFunc(const ScalarBindFuncInput& input) {
-    if (input.arguments[0]->getDataType().getLogicalTypeID() != LogicalTypeID::ANY &&
-        input.arguments[1]->dataType != ListType::getChildType(input.arguments[0]->dataType)) {
+    validateArgumentType(input.arguments);
+
+    std::vector<LogicalType> types;
+    types.push_back(input.arguments[0]->getDataType().copy());
+    types.push_back(input.arguments[1]->getDataType().copy());
+
+    resolveEmptyList(input, types);
+    resolveNulls(types);
```

Comment:
line47-60 seems duplicate in this PR. And I have a feeling it must be duplicated with other list functions as well. Can we refactor the logic and abstract common logics.

Reply:
Yes, list_concat has a similar logic. We can refactor them as well